File: benchmarks/schelling/v1/subjects/julia/tensnap.jl
# Publication environment adapter over the example's shared scenario factory.
const EXAMPLE_DIR = normpath(joinpath(@__DIR__, "../../../../../examples/julia"))
include(joinpath(EXAMPLE_DIR, "utils.jl"))
include(joinpath(EXAMPLE_DIR, "schelling.jl"))
include(joinpath(EXAMPLE_DIR, "schelling_tensnap.jl"))

SCHELLING_DYNAMICS_VERSION == 1 || error("benchmark adapter requires Schelling dynamics v1")

config = SchellingConfig(
	gridwidth = parse_env(Int, "TENSNAP_SCHELLING_WIDTH", DEFAULT_GRID_W),
	gridheight = parse_env(Int, "TENSNAP_SCHELLING_HEIGHT", DEFAULT_GRID_H),
	density = parse_env(Float64, "TENSNAP_SCHELLING_DENSITY", DEFAULT_DENSITY),
	balance = parse_env(Float64, "TENSNAP_SCHELLING_BALANCE", DEFAULT_BALANCE),
	similarity_threshold = parse_env(Float64, "TENSNAP_SCHELLING_THRESHOLD", DEFAULT_SIMILARITY_THRESHOLD),
	seed = parse_optional_env(Int, "TENSNAP_SCHELLING_SEED"),
)
server_port = parse_env(Int, "TENSNAP_SERVER_PORT", 8765)
scenario = create_schelling_scenario(
	config;
	port = server_port,
	use_msgpack = true,
	include_parameters = false,
	include_charts = false,
)

@info "Starting Julia Schelling benchmark subject" url="ws://localhost:$server_port"
TenSnap.run!(scenario)

File: benchmarks/schelling/v1/subjects/js/tensnap.ts
// Publication environment adapter over the production example WebSocket host.
import type { SchellingConfig } from '../../../../../examples/js/src/models/schelling';
import { startJsExampleWebSocketDemo } from '../../../../../examples/js/src/entries/main-ws';

function numberEnv(name: string, fallback: number): number {
  const raw = process.env[name];
  if (raw === undefined || raw === '') return fallback;
  const value = Number(raw);
  if (!Number.isFinite(value)) throw new Error(`${name} must be a finite number.`);
  return value;
}

function configFromEnvironment(): SchellingConfig {
  return {
    gridWidth: numberEnv('TENSNAP_SCHELLING_WIDTH', 50),
    gridHeight: numberEnv('TENSNAP_SCHELLING_HEIGHT', 50),
    density: numberEnv('TENSNAP_SCHELLING_DENSITY', 0.8),
    balance: numberEnv('TENSNAP_SCHELLING_BALANCE', 0.5),
    similarityThreshold: numberEnv('TENSNAP_SCHELLING_THRESHOLD', 0.7),
    seed: numberEnv('TENSNAP_SCHELLING_SEED', 7),
  };
}

async function main(): Promise<void> {
  const demo = await startJsExampleWebSocketDemo('schelling', {
    port: numberEnv('TENSNAP_SERVER_PORT', 8765),
    encoding: 'json',
    config: configFromEnvironment(),
  });
  console.log(`Started ${demo.name} at ${demo.url ?? 'unknown url'}`);
}

void main();
